### ISSUE TITLE
Infinite loop/memory-usage in irc.split()

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3446]
|<img alt="my007ms's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| my007ms reported|
|-|-|
|Trac ID|trac#3446|
|Type|defect|
|Created|2008-09-20 10:54:46Z|
|Branch|https://github.com/twisted/twisted/tree/irc-split-bug-3446|

This is the exact ircLogBot.py example
http://paste.pocoo.org/show/85757/
all i change in line 94 ,95
also i change irc nickname in line because it was long 55
after this small changes if i run this code and any one trigger line  94 ,95 from irc my machine will stop working in short time.

i have
python-twisted       8.1.0-3
python-twisted-words 8.1.0-2

<details><summary>Searchable metadata</summary>

```
trac-id__3446 3446
type__defect defect
reporter__my007ms my007ms
priority__normal normal
milestone__ 
branch__branches_irc_split_bug_3446 branches/irc-split-bug-3446
branch_author__ 
status__closed closed
resolution__fixed fixed
component__words words
keywords__ 
time__1221908086000000 1221908086000000
changetime__1222555151000000 1222555151000000
version__None None
owner__teratorn teratorn

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="teratorn's avatar" src="https://avatars.githubusercontent.com/u/154069?s=50" width="50" height="50">](https://github.com/teratorn)<a name="note_1"></a>|@teratorn set status to `closed`|
|-|-|

I talked to my007ms on IRC and he is currently unable to provide any real details as to what is happening here. 

my007ms, please start a new ticket with an appropriate Subject and as many specific details as you can. Nobody will be able to help you without a more descriptive ticket.

**Comment by twisted-trac** [comment]
|[<img alt="teratorn's avatar" src="https://avatars.githubusercontent.com/u/154069?s=50" width="50" height="50">](https://github.com/teratorn)|@teratorn set status to `reopened`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="teratorn's avatar" src="https://avatars.githubusercontent.com/u/154069?s=50" width="50" height="50">](https://github.com/teratorn)<a name="note_3"></a>|@teratorn set owner to @teratorn<br>@teratorn set status to `new`|
|-|-|

After more discussion on IRC I decided to try the provided code.. it causes 100% cpu and expanding memory usage... breaking in with WingIDE shows it to be stuck in irc.split() which is getting hung up on some weird characters in the input string.

I'll have a go at fixing this.

**Comment by twisted-trac** [comment]
|[<img alt="teratorn's avatar" src="https://avatars.githubusercontent.com/u/154069?s=50" width="50" height="50">](https://github.com/teratorn)<a name="note_4"></a>|@teratorn commented|
|-|-|

(In [24902]) Branching to 'irc-split-bug-3446'

**Comment by twisted-trac** [comment]
|[<img alt="teratorn's avatar" src="https://avatars.githubusercontent.com/u/154069?s=50" width="50" height="50">](https://github.com/teratorn)<a name="note_5"></a>|@teratorn set owner to @keturn|
|-|-|

The problem is that irc.split() can't handle finding a newline or space at the beginning of the string segment it is operating on, due to using an "and"/"or" hack for lack or a real ternary operator.

A fix, with test case, is implemented in the branch... ready for review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to @teratorn|
|-|-|

  * `testSplitSkipsStartingDelimiter` isn't named according to the coding standard
  * The docstring for `testSplitSkipsStartingDelimiter` should explain specifically what the correct behavior being tested for.  Using any form of _correct_ or a synonym in a test method docstring should be a hint that you're not being specific enough.  We **know** it's not a test to verify that the code behaves incorrectly. :)

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="teratorn's avatar" src="https://avatars.githubusercontent.com/u/154069?s=50" width="50" height="50">](https://github.com/teratorn)<a name="note_7"></a>|@teratorn set owner to @exarkun|
|-|-|

Alright, I think it looks better now.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set owner to @teratorn|
|-|-|

The test name is certainly different now...  but it should be `test_splitDelimiters`.  The docstring is also a bit more ambitious than the test method itself.  It says newlines will be skipped, but the test doesn't do anything with newlines.

**Comment by twisted-trac** [comment]
|[<img alt="teratorn's avatar" src="https://avatars.githubusercontent.com/u/154069?s=50" width="50" height="50">](https://github.com/teratorn)<a name="note_9"></a>|@teratorn set owner to @exarkun|
|-|-|

OK, done.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun set owner to @teratorn|
|-|-|

Looks fine, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="teratorn's avatar" src="https://avatars.githubusercontent.com/u/154069?s=50" width="50" height="50">](https://github.com/teratorn)<a name="note_11"></a>|@teratorn set status to `closed`|
|-|-|

(In [24932]) merge irc-split-bug-3446 (Infinite loop/memory-usage in irc.split())
Author: teratorn
Reviewer: exarkun
Fixes [#3446](https://github.com/twisted/twisted/issues/3446)
